Elaborate on shallow submodule options (#8298)
organicmaps/organicmaps: docs/INSTALL.md

- a `--shallow-submodules` option to _shallow clone_ the submodules (save ~1.3Gb) - this is suitable for a generic development if no work on submodules is planned.

> If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you want to do so shallowly, you can add `--depth=1` to the update command.

Reviewer: ```suggestion
If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you don't want to clone complete submodules, you can add `--depth=1` to the update command.
```

> If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you want to do so shallowly, you can run `git submodule foreach 'git config -f .gitmodules submodule.$sm_path.shallow true'` beforehand and/or add `--depth=1` to the update command.

Reviewer: What is the benefit of the longer foreach vs a shorter --depth=1 ? If there are no benefits, let's leave only a shorter option.

Author: @biodranik response below

Reviewer: I've googled and found that .shallow setting does a different thing https://stackoverflow.com/a/37933909

Author: OK maybe it's not necessary for this use case, I'll remove.
